Symfony/CS/Tests/Tokenizer/TokensTest.php

```diff
@@ -882,4 +882,125 @@ public function provideShortOpenTagEchoMonolithicPhpDetection()
             array("<?=' '\n?><?=' ';\n", false),
         );
     }
+
+    /**
+     * @dataProvider provideIsArray
+     * @requires PHP 5.4
+     *
+     * @param string $source   PHP code
+     * @param array  $expected ['tokenIndex'] int  Token index to use,
+     *                         ['multiline']  bool Is multiline array,
+     *                         ['short']      bool Is short array
+     */
+    public function testIsArray($source, array $expected)
```

Comment:
I could. After rewriting this thing 3 times I suggest we; @keradus, @stof and I, come up with how this should look like. Then I write the code and this thing can be over :)